lms/djangoapps/courseware/grades.py

```diff
@@ -126,9 +127,18 @@ def grade(student, request, course, keep_raw_scores=False):
     """
     Wraps "_grade" with the manual_transaction context manager just in case
     there are unanticipated errors.
+    Send a signal to update the minimum grade requirement status.
     """
     with manual_transaction():
-        return _grade(student, request, course, keep_raw_scores)
+        grade_summary = _grade(student, request, course, keep_raw_scores)
+        MIN_GRADE_REQUIREMENT_STATUS.send_robust(
+            sender=None,
+            request=request,
```

Comment:
Earlier I was doing the same but there are many old test cases for grade.py which don't have user in request. So all those test cases start failing. So with this change I have to fix those test cases as well. what do you think ?